examples/objectdetection/src/App.tsx

```diff
@@ -2,16 +2,53 @@ import * as React from "react";
 import { CameraStream } from "./CameraStream";
 import { Photo } from "./Photo";
 import { Settings } from "./Settings";
-import { NavigationContainer } from "@react-navigation/native";
+import { NavigationContainer, type RouteProp } from "@react-navigation/native";
 import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
 import { type RootTabParamList } from "./navigation";
+import Ionicons from "react-native-vector-icons/Ionicons";
 
 const Tab = createBottomTabNavigator<RootTabParamList>();
 
+type TabBarIconProps = {
+  focused: boolean;
+  color: string;
+  size: number;
+  route: RouteProp<RootTabParamList, keyof RootTabParamList>;
+};
+
+const RenderTabBarIcon: React.FC<TabBarIconProps> = ({
+  focused,
+  color,
+  size,
+  route,
+}) => {
+  let iconName;
+
+  if (route.name === "CameraStream") {
+    iconName = focused ? "camera" : "camera-outline";
+  } else if (route.name === "Photo") {
+    iconName = focused ? "document" : "document-outline";
+  } else {
+    // if (route.name === "Settings")
+    iconName = focused ? "cog" : "cog-outline";
+  }
+
+  return <Ionicons name={iconName} size={size} color={color} />;
+};
+
 function App() {
   return (
     <NavigationContainer>
-      <Tab.Navigator initialRouteName="CameraStream">
+      <Tab.Navigator
+        initialRouteName="CameraStream"
+        screenOptions={({ route }) => ({
+          tabBarIcon: ({ focused, color, size }) => {
+            return RenderTabBarIcon({ focused, color, size, route });
```

Comment:
Sorry @cdiddy77 - when you get a chance would you mind taking a look at [https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md) - there appears to be an issue with nesting "unstable" components, which is what led me down the path I took.  For instance, if I called my component with the <> syntax, it still through the error even though I pulled out the component.  I think I may mean that I'm working around the linter warning without fixing the underlying issue.

If I change the code to:
```
      <Tab.Navigator
        initialRouteName="CameraStream"
        screenOptions={({ route }) => ({
          tabBarIcon: ({ focused, color, size }) => (
            <TabBarIcon focused={focused} color={color} size={size} route={route} />
          ),
          tabBarActiveTintColor: "tomato",
          tabBarInactiveTintColor: "gray",
        })}
      >
```
I get the same linter warning.

I _think_ the issue here is that my components are rendered every time the tree is rebuilt, while optimally we'd like them to be re-rendered only if the `route` parameter changes.  But memoization, which I think should be the answer to this problem is discouraged by the above.  (This SO answer may be relevant)[https://stackoverflow.com/a/72589674/2197085] - it talks about a way to apply memoization correctly, but I can't see how to do that since hooks have to be declared at the top of the component and the thing I want to memorize it the route parameter that is being passed to the screenOptions lambda function.